### PR TITLE
Fix issue button used action-primary class changed color if click outside

### LINKED ISSUE #27095: In Floating cart delete pop-up, When user click on out side of OK button than button color is changed.
### Preconditions (*)

1. Magento2.4-develop

### Steps to reproduce (*)

1. Add to cart product
2. Goto mini cart and click on remove product

### Expected result (*)

button-solution

### Actual result (*)

button-issue

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Continue works by @neerajchand after long inactivity in PR 27560
update with correction in code style base magento standards

### Fixed Issues (if relevant)

1. fixes magento/magento2#27095: In Floating cart delete pop-up, In Floating cart delete pop-up, When user click

### Manual testing scenarios (*)

1. Magento 2.3.4+
2. Theme luma
3. Add any products to cart
4. Remove product added
5. Click outside of modal confirm remove

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by edenduong** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @edenduong. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @edenduong, here is your [new Magento instance](https://pr-28083.instances.magento-community.engineering).
Admin access: [https://pr-28083.instances.magento-community.engineering/admin_3c9c](https://pr-28083.instances.magento-community.engineering/admin_3c9c)
Login: `77939bf5` Password: `cf97839cf64e`
Instance will be terminated in up to 3 hours.

**Comment by edenduong** [comment]
Hi @mrtuvn :

Could you please solve this issue? I see it has small white color at the OK button when click outside

image

**Review by edenduong — CHANGES_REQUESTED** [review]
Hi @mrtuvn :

Could you please solve this issue? I see it has small white color at the OK button when click outside

image

**Comment by mrtuvn** [comment]
> Hi @mrtuvn :
> 
> Could you please solve this issue? I see it has small white color at the OK button when click outside
> 
> image

i think this is default of magento ! Do you meaning the white small bar in the top and bottom of button Ok ? It's is box-shadow applied to button

**Comment by edenduong** [comment]
Hi @mrtuvn :

Please see:
Before:
image

After click out side:

image

File: app/design/frontend/Magento/luma/web/css/source/_buttons.less [hunk 1]
```diff
@@ -17,8 +17,6 @@
 
     button {
         &:not(.primary) {
-            .lib-css(box-shadow, @button__shadow);
-
             &:active {
```
Comment (mrtuvn) [inline app/design/frontend/Magento/luma/web/css/source/_buttons.less:20]: @edenduong this line apply box shadow for every button doesn't have class primary. In luma theme you can see every button will have white space on top
After this change can affect to all button. I'm not sure about reason of this rule here

**Comment by edenduong** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @edenduong. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @edenduong, here is your [new Magento instance](https://pr-28083.instances.magento-community.engineering).
Admin access: [https://pr-28083.instances.magento-community.engineering/admin_c201](https://pr-28083.instances.magento-community.engineering/admin_c201)
Login: `5002a46e` Password: `4de8913c24ea`
Instance will be terminated in up to 3 hours.

**Review by edenduong — APPROVED** [review]
Thank you for your contribution !

**Comment by magento-engcom-team** [comment]
Hi @edenduong, thank you for the review.
[ENGCOM-7510](https://jira.corp.magento.com/browse/ENGCOM-7510) has been created to process this Pull Request

**Review by ptylek — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ptylek, thank you for the review.
[ENGCOM-7510](https://jira.corp.magento.com/browse/ENGCOM-7510) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

Magento 2.4-develop (**Luma** theme)

**Manual testing scenario:**

1. Add any products to the cart;
2. Click on remove button in mini-cart;
3. Click outside of modal confirm remove;

**Before:** :heavy_multiplication_x: 

Peek 2020-05-06 14-36

**After:** :heavy_check_mark: 

Peek 2020-05-06 14-45

The same results and with the **Blank** theme

**Comment by engcom-Charlie** [comment]
The failing functional tests are not related to the change in this pr.

**Pull request merged**